errors/definitions.go

```diff
@@ -0,0 +1,141 @@
+package errors
+
+import (
+	"github.com/bitcoin-sv/spv-wallet/models"
```

Comment:
Unfortunately I must say that I don't like the idea to add a dependency to models here (neither the idea to create separate package for error model and have it in dependencies 🙅 ).

My suggestion would be to choose between:
a. change models.SPVError to be some interface (add some interface that SPVError would implement). Then handle errors based on that interface in spv-wallet. And here just make the same methods as this interface has, so it should fit it out of the box.
b. or add here similar struct and handle it in similar way as SPVErrror in spv-wallet
c. or translate go-paymail errors into SPVError in spv-wallet

Reply:
Done, I chose option b. Added models to errors directory and removed models package

Reply:
After tests code as refactored to option a